lib/srv/usermgmt.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package srv
+
+import (
+	"os/user"
+	"runtime"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	"github.com/siddontang/go/log"
+)
+
+func NewUserManagement() (UserManagement, error) {
+	if runtime.GOOS == "windows" {
```

Comment:
Why only Windows? This wouldn't work on macOS either, would it? I know it doesn't keep all users in /etc/passwd.

Reply:
I'm fairly sure (but cannot verify, i dont have a mac) that getpwent will get all users, including users outside of `/etc/passwd`. At least according to the GNU man page for getpwent, it will retrieve entries from NIS and LDAP in addition to `/etc/passwd`, so MacOS may behave similarly

Everything else comes from the go standard library and should be cross platform

Reply:
For what it's worth it, macOS doesn't have `useradd`, `groupadd`, `usermod` or `userdel`, so maybe that whole file could be changed to only affect linux?